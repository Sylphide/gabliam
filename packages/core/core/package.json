{
  "name": "@gabliam/core",
  "version": "8.0.0-alpha.97+ec328fa",
  "description": "Gabliam is on top of inversify",
  "main": "lib/index.js",
  "typings": "lib/index.d.ts",
  "module": "lib/index.js",
  "scripts": {
    "build": "rimraf lib && tsc -p tsconfig.build.json"
  },
  "files": [
    "lib",
    "src"
  ],
  "dependencies": {
    "@gabliam/expression": "8.0.0-alpha.97+ec328fa",
    "@types/joi": "^14.0.1",
    "caller": "^1.0.1",
    "co": "^4.6.0",
    "debug": "^4.1.1",
    "gert": "^2.0.0",
    "gert-topo-sort": "^1.0.0",
    "glob": "^7.1.3",
    "inversify": "^5.0.1",
    "inversify-logger-middleware": "^3.0.0",
    "joi": "^14.3.1",
    "js-yaml": "^3.12.1",
    "lodash": "^4.17.11",
    "shortstop": "^1.0.3",
    "shortstop-handlers": "^1.0.1",
    "tslib": "^1.9.3"
  },
  "devDependencies": {
    "@types/debug": "^0.0.31",
    "@types/fs-extra": "^5.0.4",
    "@types/glob": "^7.1.1",
    "@types/js-yaml": "^3.11.4",
    "@types/lodash": "^4.14.119",
    "fs-extra": "^7.0.1",
    "properties": "^1.2.1",
    "rxjs": "^6.3.3",
    "toml": "^2.3.3"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/gabliam/gabliam.git"
  },
  "author": "David Touzet <dtouzet@gmail.com>",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/gabliam/gabliam/issues"
  },
  "homepage": "https://github.com/gabliam/gabliam/tree/master/packages/core#readme",
  "engines": {
    "node": ">=8.9.0"
  },
  "gitHead": "ec328fa0b62f474d470f74a2fc9637838f54091e"
}
