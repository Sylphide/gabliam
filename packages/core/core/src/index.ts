import 'reflect-metadata';
export {
  decorate,
  inject,
  injectable,
  interfaces as inversifyInterfaces,
  multiInject,
  named,
  optional,
  postConstruct,
  tagged,
  targetName,
  unmanaged,
} from 'inversify';
export {
  APP_CONFIG,
  CORE_CONFIG,
  INJECT_CONTAINER_KEY,
  VALUE_EXTRACTOR,
} from './constants';
export * from './container';
export * from './decorator';
export * from './errors';
export * from './gabliam';
export * from './interfaces';
export * from './joi';
export { configResolver, FileLoader, Resolver } from './loaders';
export * from './metadata';
export * from './promise-utils';
export * from './reflection';
export * from './registry';
export * from './utils';
