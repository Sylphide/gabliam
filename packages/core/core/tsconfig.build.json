{
  "extends": "../../../tsconfig.base",
  "compilerOptions": {
    "sourceMap": false,
    "outDir": "../../../dist/packages/core/core",
    "baseUrl": ".",
    "rootDir": ".",
    "paths": {
      "@gabliam/expression": ["../../../dist/packages/core/expression"]
    }
  },
  "include":[
    "index.ts",
    "src/**/*.ts"
  ]
}
