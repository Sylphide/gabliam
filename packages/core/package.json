{
  "name": "@gabliam/core",
  "version": "7.9.0",
  "description": "Gabliam is on top of inversify",
  "main": "lib/index.js",
  "typings": "lib/index.d.ts",
  "module": "lib/index.js",
  "scripts": {
    "build": "rimraf lib && tsc -p tsconfig.build.json"
  },
  "files": [
    "lib",
    "src"
  ],
  "dependencies": {
    "@gabliam/expression": "7.8.2",
    "@types/joi": "^13.0.8",
    "caller": "^1.0.1",
    "debug": "^3.0.0",
    "gert": "^2.0.0",
    "gert-topo-sort": "^1.0.0",
    "glob": "^7.1.1",
    "inversify": "^4.3.0",
    "inversify-logger-middleware": "^3.0.0",
    "joi": "^13.0.0",
    "js-yaml": "^3.8.1",
    "lodash": "^4.17.4",
    "shortstop": "^1.0.3",
    "shortstop-handlers": "^1.0.1",
    "tslib": "^1.8.0"
  },
  "devDependencies": {
    "@types/debug": "^0.0.30",
    "@types/glob": "^5.0.30",
    "@types/js-yaml": "^3.5.29",
    "@types/lodash": "^4.14.53"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/gabliam/gabliam.git"
  },
  "author": "David Touzet <dtouzet@gmail.com>",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/gabliam/gabliam/issues"
  },
  "homepage": "https://github.com/gabliam/gabliam/tree/master/packages/core#readme",
  "engines": {
    "node": ">=8.9.0"
  }
}
