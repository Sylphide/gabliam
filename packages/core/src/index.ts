import 'reflect-metadata';
export * from './interfaces';
export * from './decorators';
export * from './gabliam';
export {
  APP_CONFIG,
  CORE_CONFIG,
  VALUE_EXTRACTOR,
  INJECT_CONTAINER_KEY,
} from './constants';

export * from './registry';
export * from './errors';
export * from './utils';
export * from './joi';
export * from './container';
export { FileLoader, Resolver, configResolver } from './loaders';

export {
  injectable,
  tagged,
  named,
  inject,
  optional,
  unmanaged,
  multiInject,
  targetName,
  decorate,
  postConstruct,
  interfaces as inversifyInterfaces,
} from 'inversify';
