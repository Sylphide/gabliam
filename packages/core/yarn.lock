# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
# yarn lockfile v1


"@types/debug@^0.0.30":
  version "0.0.30"
  resolved "https://registry.yarnpkg.com/@types/debug/-/debug-0.0.30.tgz#dc1e40f7af3b9c815013a7860e6252f6352a84df"

"@types/glob@^5.0.30":
  version "5.0.32"
  resolved "https://registry.yarnpkg.com/@types/glob/-/glob-5.0.32.tgz#aec5cfe987c72f099fdb1184452986aa506d5e8f"
  dependencies:
    "@types/minimatch" "*"
    "@types/node" "*"

"@types/joi@^10.0.1":
  version "10.4.3"
  resolved "https://registry.yarnpkg.com/@types/joi/-/joi-10.4.3.tgz#1d82cd9a79194f3dd0685d5d2c220a77a9c0776b"

"@types/js-yaml@^3.5.29":
  version "3.9.1"
  resolved "https://registry.yarnpkg.com/@types/js-yaml/-/js-yaml-3.9.1.tgz#2f3c142771bb345829ce690c5838760b6b9ba553"

"@types/lodash@^4.14.53":
  version "4.14.76"
  resolved "https://registry.yarnpkg.com/@types/lodash/-/lodash-4.14.76.tgz#87874f766774d54e89589697340be9496fb8bf70"

"@types/minimatch@*":
  version "2.0.29"
  resolved "https://registry.yarnpkg.com/@types/minimatch/-/minimatch-2.0.29.tgz#5002e14f75e2d71e564281df0431c8c1b4a2a36a"

"@types/node@*":
  version "8.0.20"
  resolved "https://registry.yarnpkg.com/@types/node/-/node-8.0.20.tgz#65c7375255c24b184c215a5d0b63247c32f01c91"

ansi-regex@^2.0.0:
  version "2.1.1"
  resolved "https://registry.yarnpkg.com/ansi-regex/-/ansi-regex-2.1.1.tgz#c3b33ab5ee360d86e0e628f0468ae7ef27d654df"

ansi-styles@^2.2.1:
  version "2.2.1"
  resolved "https://registry.yarnpkg.com/ansi-styles/-/ansi-styles-2.2.1.tgz#b432dd3358b634cf75e1e4664368240533c1ddbe"

argparse@^1.0.7:
  version "1.0.9"
  resolved "https://registry.yarnpkg.com/argparse/-/argparse-1.0.9.tgz#73d83bc263f86e97f8cc4f6bae1b0e90a7d22c86"
  dependencies:
    sprintf-js "~1.0.2"

async@~0.2.10:
  version "0.2.10"
  resolved "https://registry.yarnpkg.com/async/-/async-0.2.10.tgz#b6bbe0b0674b9d719708ca38de8c237cb526c3d1"

balanced-match@^1.0.0:
  version "1.0.0"
  resolved "https://registry.yarnpkg.com/balanced-match/-/balanced-match-1.0.0.tgz#89b4d199ab2bee49de164ea02b89ce462d71b767"

brace-expansion@^1.1.7:
  version "1.1.8"
  resolved "https://registry.yarnpkg.com/brace-expansion/-/brace-expansion-1.1.8.tgz#c07b211c7c952ec1f8efd51a77ef0d1d3990a292"
  dependencies:
    balanced-match "^1.0.0"
    concat-map "0.0.1"

caller@^1.0.1:
  version "1.0.1"
  resolved "https://registry.yarnpkg.com/caller/-/caller-1.0.1.tgz#b851860f70e195db3d277395aa1a7e23ea30ecf5"

chalk@^1.1.3:
  version "1.1.3"
  resolved "https://registry.yarnpkg.com/chalk/-/chalk-1.1.3.tgz#a8115c55e4a702fe4d150abd3872822a7e09fc98"
  dependencies:
    ansi-styles "^2.2.1"
    escape-string-regexp "^1.0.2"
    has-ansi "^2.0.0"
    strip-ansi "^3.0.0"
    supports-color "^2.0.0"

concat-map@0.0.1:
  version "0.0.1"
  resolved "https://registry.yarnpkg.com/concat-map/-/concat-map-0.0.1.tgz#d8a96bd77fd68df7793a73036a3ba0d5405d477b"

core-util-is@^1.0.1, core-util-is@~1.0.1:
  version "1.0.2"
  resolved "https://registry.yarnpkg.com/core-util-is/-/core-util-is-1.0.2.tgz#b5fd54220aa2bc5ab57aab7140c940754503c1a7"

debug@^3.0.0:
  version "3.1.0"
  resolved "https://registry.yarnpkg.com/debug/-/debug-3.1.0.tgz#5bb5a0672628b64149566ba16819e61518c67261"
  dependencies:
    ms "2.0.0"

escape-string-regexp@^1.0.2:
  version "1.0.5"
  resolved "https://registry.yarnpkg.com/escape-string-regexp/-/escape-string-regexp-1.0.5.tgz#1b61c0562190a8dff6ae3bb2cf0200ca130b86d4"

esprima@^4.0.0:
  version "4.0.0"
  resolved "https://registry.yarnpkg.com/esprima/-/esprima-4.0.0.tgz#4499eddcd1110e0b218bacf2fa7f7f59f55ca804"

fs.realpath@^1.0.0:
  version "1.0.0"
  resolved "https://registry.yarnpkg.com/fs.realpath/-/fs.realpath-1.0.0.tgz#1504ad2523158caa40db4a2787cb01411994ea4f"

gert-topo-sort@^1.0.0:
  version "1.0.0"
  resolved "https://registry.yarnpkg.com/gert-topo-sort/-/gert-topo-sort-1.0.0.tgz#e63d78cb134576c43cda2116f1689bce93a8dac0"

gert@^2.0.0:
  version "2.0.0"
  resolved "https://registry.yarnpkg.com/gert/-/gert-2.0.0.tgz#634c649751ddd4b6f1aefb46d31f8057a1a62b71"
  dependencies:
    hoek "^2.16.3"
    joi "^6.9.1"

glob@^7.0.5, glob@^7.1.1:
  version "7.1.2"
  resolved "https://registry.yarnpkg.com/glob/-/glob-7.1.2.tgz#c19c9df9a028702d678612384a6552404c636d15"
  dependencies:
    fs.realpath "^1.0.0"
    inflight "^1.0.4"
    inherits "2"
    minimatch "^3.0.4"
    once "^1.3.0"
    path-is-absolute "^1.0.0"

has-ansi@^2.0.0:
  version "2.0.0"
  resolved "https://registry.yarnpkg.com/has-ansi/-/has-ansi-2.0.0.tgz#34f5049ce1ecdf2b0649af3ef24e45ed35416d91"
  dependencies:
    ansi-regex "^2.0.0"

hoek@2.x.x, hoek@^2.16.3:
  version "2.16.3"
  resolved "https://registry.yarnpkg.com/hoek/-/hoek-2.16.3.tgz#20bb7403d3cea398e91dc4710a8ff1b8274a25ed"

hoek@4.x.x:
  version "4.2.0"
  resolved "https://registry.yarnpkg.com/hoek/-/hoek-4.2.0.tgz#72d9d0754f7fe25ca2d01ad8f8f9a9449a89526d"

inflight@^1.0.4:
  version "1.0.6"
  resolved "https://registry.yarnpkg.com/inflight/-/inflight-1.0.6.tgz#49bd6331d7d02d0c09bc910a1075ba8165b56df9"
  dependencies:
    once "^1.3.0"
    wrappy "1"

inherits@2:
  version "2.0.3"
  resolved "https://registry.yarnpkg.com/inherits/-/inherits-2.0.3.tgz#633c2c83e3da42a502f52466022480f4208261de"

inversify-logger-middleware@^3.0.0:
  version "3.0.0"
  resolved "https://registry.yarnpkg.com/inversify-logger-middleware/-/inversify-logger-middleware-3.0.0.tgz#ded2c8c6055828255cd599dfec68faf1eb999ec3"
  dependencies:
    chalk "^1.1.3"

inversify@^4.3.0:
  version "4.3.0"
  resolved "https://registry.yarnpkg.com/inversify/-/inversify-4.3.0.tgz#d2ecd2ae18340e7f1ab5148a3e44b87080391366"

isemail@1.x.x:
  version "1.2.0"
  resolved "https://registry.yarnpkg.com/isemail/-/isemail-1.2.0.tgz#be03df8cc3e29de4d2c5df6501263f1fa4595e9a"

isemail@3.x.x:
  version "3.0.0"
  resolved "https://registry.yarnpkg.com/isemail/-/isemail-3.0.0.tgz#c89a46bb7a3361e1759f8028f9082488ecce3dff"
  dependencies:
    punycode "2.x.x"

joi@^11.0.1:
  version "11.1.1"
  resolved "https://registry.yarnpkg.com/joi/-/joi-11.1.1.tgz#08194537d3b9c796bac8666deb3a825c1a5d83c1"
  dependencies:
    hoek "4.x.x"
    isemail "3.x.x"
    topo "2.x.x"

joi@^6.9.1:
  version "6.10.1"
  resolved "https://registry.yarnpkg.com/joi/-/joi-6.10.1.tgz#4d50c318079122000fe5f16af1ff8e1917b77e06"
  dependencies:
    hoek "2.x.x"
    isemail "1.x.x"
    moment "2.x.x"
    topo "1.x.x"

js-yaml@^3.8.1:
  version "3.10.0"
  resolved "https://registry.yarnpkg.com/js-yaml/-/js-yaml-3.10.0.tgz#2e78441646bd4682e963f22b6e92823c309c62dc"
  dependencies:
    argparse "^1.0.7"
    esprima "^4.0.0"

lodash@^4.17.4:
  version "4.17.4"
  resolved "https://registry.yarnpkg.com/lodash/-/lodash-4.17.4.tgz#78203a4d1c328ae1d86dca6460e369b57f4055ae"

minimatch@^3.0.4:
  version "3.0.4"
  resolved "https://registry.yarnpkg.com/minimatch/-/minimatch-3.0.4.tgz#5166e286457f03306064be5497e8dbb0c3d32083"
  dependencies:
    brace-expansion "^1.1.7"

moment@2.x.x:
  version "2.18.1"
  resolved "https://registry.yarnpkg.com/moment/-/moment-2.18.1.tgz#c36193dd3ce1c2eed2adb7c802dbbc77a81b1c0f"

ms@2.0.0:
  version "2.0.0"
  resolved "https://registry.yarnpkg.com/ms/-/ms-2.0.0.tgz#5608aeadfc00be6c2901df5f9861788de0d597c8"

once@^1.3.0:
  version "1.4.0"
  resolved "https://registry.yarnpkg.com/once/-/once-1.4.0.tgz#583b1aa775961d4b113ac17d9c50baef9dd76bd1"
  dependencies:
    wrappy "1"

path-is-absolute@^1.0.0:
  version "1.0.1"
  resolved "https://registry.yarnpkg.com/path-is-absolute/-/path-is-absolute-1.0.1.tgz#174b9268735534ffbc7ace6bf53a5a9e1b5c5f5f"

punycode@2.x.x:
  version "2.1.0"
  resolved "https://registry.yarnpkg.com/punycode/-/punycode-2.1.0.tgz#5f863edc89b96db09074bad7947bf09056ca4e7d"

shortstop-handlers@^1.0.1:
  version "1.0.1"
  resolved "https://registry.yarnpkg.com/shortstop-handlers/-/shortstop-handlers-1.0.1.tgz#0755acb6604c76c9c0e76a9e943f93790bab1dc4"
  dependencies:
    caller "^1.0.1"
    core-util-is "^1.0.1"
    glob "^7.0.5"

shortstop@^1.0.3:
  version "1.0.3"
  resolved "https://registry.yarnpkg.com/shortstop/-/shortstop-1.0.3.tgz#d43769c36fee7e20a3b9bfd2d0878935ff86e7c6"
  dependencies:
    async "~0.2.10"
    core-util-is "~1.0.1"

sprintf-js@~1.0.2:
  version "1.0.3"
  resolved "https://registry.yarnpkg.com/sprintf-js/-/sprintf-js-1.0.3.tgz#04e6926f662895354f3dd015203633b857297e2c"

strip-ansi@^3.0.0:
  version "3.0.1"
  resolved "https://registry.yarnpkg.com/strip-ansi/-/strip-ansi-3.0.1.tgz#6a385fb8853d952d5ff05d0e8aaf94278dc63dcf"
  dependencies:
    ansi-regex "^2.0.0"

supports-color@^2.0.0:
  version "2.0.0"
  resolved "https://registry.yarnpkg.com/supports-color/-/supports-color-2.0.0.tgz#535d045ce6b6363fa40117084629995e9df324c7"

topo@1.x.x:
  version "1.1.0"
  resolved "https://registry.yarnpkg.com/topo/-/topo-1.1.0.tgz#e9d751615d1bb87dc865db182fa1ca0a5ef536d5"
  dependencies:
    hoek "2.x.x"

topo@2.x.x:
  version "2.0.2"
  resolved "https://registry.yarnpkg.com/topo/-/topo-2.0.2.tgz#cd5615752539057c0dc0491a621c3bc6fbe1d182"
  dependencies:
    hoek "4.x.x"

wrappy@1:
  version "1.0.2"
  resolved "https://registry.yarnpkg.com/wrappy/-/wrappy-1.0.2.tgz#b5243d8f3ec1aa35f1364605bc0d1036e30ab69f"
