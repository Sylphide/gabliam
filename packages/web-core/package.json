{
  "name": "@gabliam/web-core",
  "version": "7.9.4",
  "description": "Gabliam plugin for add web-core",
  "keywords": [
    "gabliam",
    "gabliam-plugin",
    "plugin",
    "web-core",
    "rest"
  ],
  "main": "lib/index.js",
  "typings": "lib/index.d.ts",
  "module": "lib/index.js",
  "scripts": {
    "build": "rimraf lib && tsc -p tsconfig.build.json"
  },
  "files": [
    "lib",
    "src"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://github.com/gabliam/gabliam.git"
  },
  "author": "David Touzet <dtouzet@gmail.com>",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/gabliam/gabliam/issues"
  },
  "homepage": "https://github.com/gabliam/gabliam/tree/master/packages/web-core#readme",
  "peerDependencies": {
    "@gabliam/core": "^7.2.0"
  },
  "devDependencies": {
    "@gabliam/core": "7.9.4",
    "@types/escape-html": "^0.0.20"
  },
  "dependencies": {
    "escape-html": "^1.0.3",
    "http-status-codes": "^1.2.0",
    "tslib": "^1.8.0"
  },
  "engines": {
    "node": ">=8.9.0"
  }
}
