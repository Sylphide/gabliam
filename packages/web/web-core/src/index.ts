export * from './constants';
export * from './cookies';
export * from './decorators';
export * from './errors';
export * from './execution-context';
export * from './gab-context';
export * from './gab-request';
export * from './gab-response';
export * from './interface';
export * from './plugin-config';
export * from './response-entity';
export * from './utils';
export * from './validate';
export * from './web-configuration';
export * from './web-plugin-base';
