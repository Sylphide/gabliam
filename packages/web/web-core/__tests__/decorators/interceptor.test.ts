import { reflection } from '@gabliam/core';
import { UseInterceptors } from '@gabliam/web-core';

describe('interceptors decorators', () => {
  test('should add Bean metadata to a class when decorated with @UseInterceptors', () => {
    class TestMiddleware {
      @UseInterceptors('test')
      testMethod() {}

      @UseInterceptors('test2')
      test2Method() {}
    }

    expect(reflection.propMetadata(TestMiddleware)).toMatchSnapshot();
  });
});
