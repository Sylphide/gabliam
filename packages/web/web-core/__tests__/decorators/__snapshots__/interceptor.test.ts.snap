// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`interceptors decorators should add Bean metadata to a class when decorated with @UseInterceptors 1`] = `
Object {
  "test2Method": Array [
    DecoratorFactory {
      "ids": Array [
        "test2",
      ],
    },
  ],
  "testMethod": Array [
    DecoratorFactory {
      "ids": Array [
        "test",
      ],
    },
  ],
}
`;
